src/nvim/eval.c

```diff
@@ -12245,19 +12249,18 @@ static void f_reltime(typval_T *argvars, typval_T *rettv) FUNC_ATTR_NONNULL_ALL
   // (varnumber_T is defined as int). For all our supported platforms, int's
   // are at least 32-bits wide. So we'll use two 32-bit values to store it.
   union {
-    struct { varnumber_T low, high; } split;
+    struct { uint32_t low, high; } split;
     proftime_T prof;
   } u = { .prof = res };
 
-  // statically assert that the union type conv will provide the correct
-  // results, if varnumber_T or proftime_T change, the union cast will need
-  // to be revised.
-  STATIC_ASSERT(sizeof(u.prof) == sizeof(u) && sizeof(u.split) == sizeof(u),
-      "type punning will produce incorrect results on this platform");
```

Comment:
I removed the original `STATIC_ASSERT` because we use a `uint64_t` for time and the change to using `uint32_t` for the high and low bits made it redundant. Asserting that `sizeof(varnumber_T) >= sizeof(uint32_t)` sounds ok, though do we, or will we, support platforms where this is not true?